src/components/AddItemForm.tsx

```diff
@@ -1,7 +1,19 @@
-import React from 'react';
+import React, { useState } from 'react';
 
-function AddItemForm() {
+interface IProps {
+  handleAdd: (textTask: string) => void;
+}
+
+function AddItemForm(props: IProps) {
+  const { handleAdd } = props;
   const domInputId = 'new-todo-input';
```

Comment:
[Sugestão]
Para simplificar o uso e reduzir uma linha o code poderia ficar assim:

```suggestion
function AddItemForm({ handleAdd }: IProps) {
  const domInputId = 'new-todo-input';
  // Agora é possível utilizar o nome handleAdd dentro da função normalmente!
```

Reply:
Ah certo, mas tenho uma dúvida em relação de boas práticas que é a seguinte: caso exista vários parametros nesse componentes, posso fazer direto na function ou é melhor criar um const e desestruturar a props igual fiz inicialmente?

Reply:
Acho que de qualquer forma o aberto vai valer mais a pena, a interface e o autocomplete já vão demarcar bem a interface do componente.

Mas a sua escrita ainda é válida, mas acho que vai ser em situações mais como:
Imagine uma função que recebe como parametros:
- um objeto de configurações
- uma string a ser processada
- um booleano definindo um modificador

Aqui é interessante tratar este objeto de configurações e consumílo como você fez. Uma vez que, neste contexto, existe uma separação lógica clara entre cada paramêtro da função e configurações foi simplificado para um objeto único pois define um contexto para aquele dado.

Reply:
Ah certo, deu para entender. Valeu pela explicação.